### ISSUE TITLE
braid update to older commits

### ISSUE THREAD

**Issue by tirolerstefan** [cristibalan/braid#32]
I tried to step back to an earlier commit with "braid update --revision" but once I used "braid update --head" I cannot lock it to older revisions or I could not find out how (squashed = true).

The message I get (1.0.1):
Braid: Mirror 'sub' is already up to date.

Is this possible somehow or restricted intentionally?

**Comment by realityforge** [comment]
It is not restricted intentionally. That seems like a bug...

**Comment by tirolerstefan** [comment]
It's working when I comment lines 44-47 in update.rb:

```
        new_revision    = validate_new_revision(mirror, options['revision'])
        target_revision = determine_target_revision(new_revision)

        #if mirror.merged?(target_revision)
          #msg "Mirror '#{mirror.path}' is already up to date."
          #return
        #end

        if mirror.squashed?
          diff          = mirror.diff
          base_revision = mirror.base_revision
        end
```

But I don't know the background of "mirror.merged", so I don't dare to create a PR.

**Comment by realityforge** [comment]
If inally got around to looking at this and have applied a fix and pushed out version 1.0.3

Thanks for the heads up
